src/selectors.js

```diff
@@ -0,0 +1,129 @@
+        switch (settings.percentagesPosition) {
+            case 'Last column': {
+                selectors = {
+                    mark: 'td:nth-child(2) > input',
+                    weight: 'td:nth-child(4) > input',
+                    denominator: 'td:nth-child(5) > input',
+                    percentage: 'td:nth-child(6)'
+                };
+                break;
+            }
+            case 'Second column': {
+                selectors = {
+                    mark: 'td:nth-child(3) > input',
+                    weight: 'td:nth-child(5) > input',
+                    denominator: 'td:nth-child(6) > input',
+                    percentage: 'td:nth-child(2)'
+                };
+                break;
+            }
+            default: {
+                throw new Error('Unknown percentages position', settings.percentagesPosition);
+            }
+        }
+    } else if (settings.betterTableLayout) {
+        if (!settings.percentages) {
+            selectors = {
+                mark: 'td:nth-child(2) > span:first',
+                weight: 'td:nth-child(3)',
+                denominator: 'td:nth-child(2) > span:last'
+            };
+            return;
+        }
+        switch (settings.percentagesPosition) {
+            case 'Last column': {
+                selectors = {
+                    mark: 'td:nth-child(2) > span:first',
+                    weight: 'td:nth-child(3)',
+                    denominator: 'td:nth-child(2) > span:last',
+                    percentage: 'td:nth-child(5)'
+                };
+                break;
+            }
+            case 'Second column': {
+                selectors = {
+                    mark: 'td:nth-child(3) > span:first',
+                    weight: 'td:nth-child(4)',
+                    denominator: 'td:nth-child(3) > span:last',
+                    percentage: 'td:nth-child(2)'
+                };
+                break;
+            }
+            default: {
+                throw new Error('Unknown percentages position', settings.percentagesPosition);
+            }
+        }  
+    } else if (settings.percentages) {
+        switch (settings.percentagesPosition) {
+            case 'Last column': {
+                selectors = {
+                    mark: 'td:nth-child(2)',
+                    weight: 'td:nth-child(4)',
+                    denominator: 'td:nth-child(5)',
+                    percentage: 'td:nth-child(6)'
+                };
+                break;
+            }
+            case 'Second column': {
+                selectors = {
+                    mark: 'td:nth-child(3)',
+                    weight: 'td:nth-child(5)',
+                    denominator: 'td:nth-child(6)',
+                    percentage: 'td:nth-child(2)'
+                };
+                break;
+            }
+            default: {
+                throw new Error('Unknown percentages position', settings.percentagesPosition);
+            }
+        }
+    }
+};
```

Comment:
This logic is way too long and confusing, try having logic for each selector on its own (mark, weight, denominator, percentage) since many of the selectors repeat themselves.